shopfloor/actions/data.py

```diff
@@ -86,14 +90,28 @@ def package(self, record, picking=None, with_packaging=False, **kw):
     def packages(self, records, picking=None, **kw):
         return [self.package(rec, picking=picking, **kw) for rec in records]
 
-    @property
-    def _package_parser(self):
-        return [
+    def _package_parser(self, picking=None):
+        parser = [
             "id",
             "name",
-            "pack_weight:weight",
             ("package_storage_type_id:storage_type", ["id", "name"]),
         ]
+        if picking:
```

Comment:
to avoid duplicating the whole block, I would do this:

```python
    context_kwargs = {}
    if picking:
        context_kwargs["picking_id"] = picking.id
    parser.append(
        (
            "pack_weight:weight",
            lambda rec, fname, picking=picking: rec.pack_weight
            or rec.with_context(**context_kwargs).estimated_pack_weight,
         ),
    )
```

Reply:
Done